### DISCUSSION TITLE
Default drawer open, but allow closing

### DISCUSSION THREAD

**Discussion by JacobCoffee** [saadeghi/daisyui#2535] [Q&A] [answered]
Hi, 

I'd like to default a `drawer-open` on `lg` to `md` screens, but it seems that this will force these open in this view sizes no matter what. 

Is it possible to have this, but allow a button to close it without doing a bunch of extra JS?

**Comment by saadeghi** [comment, marked as answer]
In Tailwind, you can target a specific range like this `md:max-lg:`:   
https://tailwindcss.com/docs/responsive-design#targeting-a-breakpoint-range

In this example, drawer is open only from md to lg (because of `md:max-lg:drawer-open` class):   
https://play.tailwindcss.com/NZs2nX5jAb?size=1200x720

**Reply by JacobCoffee** [reply]
I'm sorry if I misunderstand but I have this, but what I am not able to do is close a drawer when in md/lg on demand

**Reply by saadeghi** [reply]
You can check the drawer checkbox using JS and control it the way you want.  

Right now, it is forced open in md,lg size. It is closable in other sizes. If you want it to be closable in md, lg sizes as well, you won't need the `drawer-open` class in the first place. Because it's either closable or not closable. I'm not sure what's your expected behavior 🤔

**Reply by JacobCoffee** [reply]
haha sorry if I'm confusing.. not really a front end guy. I was hoping to have it default open with `drawer-open`, and the drawer has a `x` button to close it if desired, or if the viewport is shrunk beloe `md`/`lg` there will be another button to open/close as desired as well

**Reply by JacobCoffee** [reply]
I may be hacking at it but I am basically using this as a collapsible full-time nav in all viewport sizes.. 
and I can do the JS fix you suggest! I was just curious if there was a native way :) Appreciate the help.

**Reply by saadeghi** [reply]
I guess you want the drawer to be visible on all sizes and changes the width (icon only / icon + text)  
In that case, use `drawer-open` class to keep it open, then you can put a button 
That button can change the width class of the sidebar (to make it go wide or narrow) using JS and also can add `hidden` class to sidebar items text for when you click it.

This way you will have sidebar all the time, and a button can make it narrow or wide.
